### PR TITLE
HB-06

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**

### CODE REVIEW

**Review by SetMyDream — APPROVED** [review]
good job!

File: src/main/java/com/cinema/Main.java [hunk 1]
```diff
@@ -62,17 +63,30 @@ public static void main(String[] args) throws AuthenticationException {
...
+        authenticationService.register(onic.getEmail(), onic.getPassword());
+
+        authenticationService.login(onic.getEmail(), onic.getPassword());
+        System.out.println(userService.findByEmail(onic.getEmail()).get());
+        User foundUser = userService.findByEmail("onic@u.com").get();
+        System.out.println("My found user: " + foundUser);
+
+        ShoppingCartService shoppingCartService =
+                (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+        shoppingCartService.registerNewShoppingCart(panas);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/Main.java]: let's move this login to authenticationService.register()

File: src/main/java/com/cinema/model/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+@Table(name = "shopping_cart")
+public class ShoppingCart {
+    @Id
+    @Column(name = "cart_id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    @OneToOne
+    private User user;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/model/ShoppingCart.java]: let's use mapsId so we will not have duplications

File: src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> getCartByUserQuery
+                    = session.createQuery("FROM ShoppingCart sc left join fetch sc.tickets "
+                    + "WHERE sc.user = : user", ShoppingCart.class);
+            getCartByUserQuery.setParameter("user", user);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java]: do you have access to user field of this shopping cart when the session is closed?

File: src/main/java/com/cinema/Main.java (hunk 1 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/Main.java]: Done

File: src/main/java/com/cinema/model/ShoppingCart.java (hunk 2 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/model/ShoppingCart.java]: Done

File: src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java (hunk 3 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java]: Done

Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java]: Now I have

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**